Fix RTL causes date in about window to be malformed, #4969

low-batt opened the pull request:
This commit will:
- Change the `Text Direction` to be `Left To Right` for the `Date` text field in `AboutWindowController.xib`
- Change the `Mirror` attribute to be `Never` for the `Date` text field in `AboutWindowController.xib`
- Change the type of the `buildDate` property in `InfoDictionary` to be `Date?` instead of `String?`
- Change `windowDidLoad` in `AboutWindowController` to format `buildDate` as a localized string
- Change `logBuildDetails` in `AppDelegate` to format `buildDate` using the `en_US` locale

These changes correct the localized build date displayed in the about window. They also change the build date displayed in the log file to not use the user's locale.

- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #4969.

---

**Description:**

low-batt committed:
Fix RTL causes date in about window to be malformed, #4969

This commit will:
- Change the Text Direction to be Left To Right for the Date text field
  in AboutWindowController.xib
- Change the Mirror attribute to be Never for the Date text field in
  AboutWindowController.xib
- Change the type of the buildDate property in InfoDictionary to be
  Date? instead of String?
- Change windowDidLoad in AboutWindowController to format buildDate as
  a localized string
- Change logBuildDetails in AppDelegate to format buildDate using the
  en_US locale

These changes correct the localized build date displayed in the about
window. They also change the build date displayed in the log file to not
use the user's locale.

ShlomoCode commented:
Additional issues in the About window in the "Contributors" tab (the arrows should be reversed like in https://github.com/iina/iina/pull/4958, and the line "powered by Crowdin" should be localizable):
![CleanShot 2024-06-02 at 04 19 12@2x](https://github.com/iina/iina/assets/78599753/a501486c-1450-49c2-abfa-8497c41b6308)

This should also be localized:
![CleanShot 2024-06-02 at 04 43 42@2x](https://github.com/iina/iina/assets/78599753/7a8a9d6b-b538-4b57-8809-15630b9763eb)

uiryuu commented:
> This should also be localized:
CleanShot 2024-06-02 at 04 43 42@2x

It's not worthy to maintain the localization for this. The only valid string to localize is "Website", but to maintain this file needs a lot of effort.

ShlomoCode commented:
Why not translate for example "The modern video player for macOS"?

uiryuu commented:
It's the slogan of the app, which is fairly important, but we cannot verify and guarantee that every localization is good. Plus translating it is very subjective, people might disagree on it.

low-batt committed:
Fix RTL causes date in about window to be malformed, #4969

This commit will:
- Change the Mirror attribute to be Never for the Date text field in
  AboutWindowController.xib
- Change the type of the buildDate property in InfoDictionary to be
  Date? instead of String?
- Change windowDidLoad in AboutWindowController to format buildDate as
  a localized string
- Change logBuildDetails in AppDelegate to format buildDate using the
  en_US locale

These changes correct the localized build date displayed in the about
window. They also change the build date displayed in the log file to not
use the user's locale.

low-batt commented:
This is what the about window looks like with the latest commit:
![about](https://github.com/iina/iina/assets/86170219/163a8020-81bf-499a-8077-cd056a015a82)

uiryuu commented:
A last confirmation from @ShlomoCode before we can merge this...

ShlomoCode commented:
yes it's normal now
I don't really like the time format when it's PM, I think a time in 24 hour format is more readable, but I understand it's Apple's design?
